File: AIAndroid/app/src/main/kotlin/ai/ecoinference/app/tools/ImageStore.kt
package ai.ecoinference.app.tools

/**
 * Holds the most recently user-attached image bytes for use by `edit_image`.
 * Set from ChatScreen when the user picks or removes a photo.
 *
 * Thread-safe via @Volatile — writes happen on the main thread,
 * reads happen on Dispatchers.Default inside PythonRunner.
 */
object ImageStore {
    @Volatile var currentImageBytes: ByteArray? = null
}

File: AIiOS/AIiOS/Services/ImageStore.swift
import UIKit

/// Holds the most recently user-attached image so the `edit_image` tool
/// can access it without the model needing to pass base64 in its call.
///
/// ChatView writes to `currentImage` whenever the user picks a photo.
/// ImageEditTools reads `currentImageData` to pass the image to Python.
final class ImageStore {

    static let shared = ImageStore()
    private init() {}

    /// The most recently attached image.  Setting this also updates
    /// `currentImageData` (JPEG, 90% quality).
    var currentImage: UIImage? {
        didSet {
            currentImageData = currentImage?.jpegData(compressionQuality: 0.9)
        }
    }

    /// JPEG-encoded bytes of `currentImage`, ready for base64 encoding.
    private(set) var currentImageData: Data? = nil
}
